test/src/unit-cppapi-deletes.cc

```diff
@@ -1526,3 +1543,381 @@ TEST_CASE_METHOD(
   remove_sparse_array();
 }
 #endif
+
+TEST_CASE_METHOD(
+    DeletesFx,
+    "CPP API: Deletion of fragment writes",
+    "[cppapi][deletes][fragments]") {
+  remove_sparse_array();
+
+  // Write fragments at timestamps 1, 3, 5, 7
+  create_sparse_array();
+  write_sparse({0, 1, 2, 3}, {1, 1, 1, 2}, {1, 2, 4, 3}, 1);
+  write_sparse({0, 1, 2, 3}, {1, 1, 1, 2}, {1, 2, 4, 3}, 3);
+  write_sparse({0, 1, 2, 3}, {1, 1, 1, 2}, {1, 2, 4, 3}, 5);
+  write_sparse({0, 1, 2, 3}, {1, 1, 1, 2}, {1, 2, 4, 3}, 7);
+  std::string commit_dir = tiledb::test::get_commit_dir(SPARSE_ARRAY_NAME);
+  std::vector<std::string> commits{vfs_.ls(SPARSE_ARRAY_NAME)};
+
+  // Conditionally consolidate and vacuum
+  bool consolidate = GENERATE(true, false);
+  bool vacuum = GENERATE(true, false);
+  if (!consolidate && vacuum) {
+    return;
+  }
+
+  if (consolidate) {
+    consolidate_commits_sparse(vacuum);
+    CHECK(tiledb::test::num_fragments(SPARSE_ARRAY_NAME) == 4);
+    commits = vfs_.ls(commit_dir);
+    if (vacuum) {
+      CHECK(commits.size() == 1);
+      CHECK(tiledb::sm::utils::parse::ends_with(
+          commits[0], tiledb::sm::constants::con_commits_file_suffix));
+    } else {
+      CHECK(commits.size() == 5);
+      bool con_exists = false;
+      for (auto commit : commits) {
+        if (tiledb::sm::utils::parse::ends_with(
+                commit, tiledb::sm::constants::con_commits_file_suffix)) {
+          con_exists = true;
+        }
+      }
+      CHECK(con_exists);
+    }
+  }
+
+  // Delete fragments
+  std::unique_ptr<Array> array =
+      std::make_unique<Array>(ctx_, SPARSE_ARRAY_NAME, TILEDB_MODIFY_EXCLUSIVE);
+  array->delete_fragments(SPARSE_ARRAY_NAME, 2, 6);
+  CHECK(tiledb::test::num_fragments(SPARSE_ARRAY_NAME) == 2);
+  array->close();
+
+  // Check commits directory after deletion
+  if (consolidate) {
```

Comment:
Could we make this a helper API that lists the commits directory and returns the number of wrt, ign, con, del, upd files? (Probably as a small class)